### ISSUE TITLE
Service looped back to same pod is not accessible for IPv6

### ISSUE THREAD

**Issue by fristonio** [cilium/cilium#13420]
Earlier this was fixed for IPv4 services. IPv6 services still have this issue.
From https://github.com/cilium/cilium/pull/243

> Whenever a service is looped back into the same pod, the IP will be used as the source. Because we are not rewriting source ports to be unique, the source IP may not be used by anyone else in the context of that host.

**Comment by fristonio** [comment]
The PR for the fix is there in https://github.com/cilium/cilium/pull/14138
I also had a local patch(might need to revisit this) that addresses the comments suggested by @pchaigno. The issue I was having was with the failure of upgrade tests possibly due to the introduction of a new tail call. As discussed on slack we decided to wait on #13032 that can possibly fix the issue. I will open a thread to discuss this on slack.

**Comment by julianwiedmann** [comment]
Closing in favor of https://github.com/cilium/cilium/issues/26733.

**Comment by karry-yky** [comment]
May I ask why the fix for this bug was eventually abandoned? Was it due to the introduction of a new tail call interfering with the functionality? If so, could you clarify which tail call specifically caused the problem?

**Comment by pchaigno** [comment]
@karry-yky We had issues with new tail calls on up/downgrades. They would cause some packet drops. That should be fixed now so anyone willing can proceed with a fix.

**Comment by karry-yky** [comment]
@pchaigno Thanks a lot for the update! Glad to hear the tail call issue is fixed. I'll go ahead and proceed with the fix. Appreciate your help!
